lib/renderer.js

```diff
@@ -83,17 +89,24 @@ class Renderer extends MarkedRenderer {
 
   // Prepend root to image path
   image(href, title, text) {
-    const { options } = this;
-
-    if (!parse(href).hostname && !options.config.relative_link
-      && options.prependRoot) {
-      href = url_for.call(options, href);
+    const { hexo, options } = this;
+    const { relative_link } = hexo.config;
+    const { lazyload, prependRoot, postId } = options;
+
+    if (!/^(#|\/\/|http(s)?:)/.test(href) && !relative_link && prependRoot) {
+      if (!/^(\/|\\)/.test(href) && postId) {
```

Comment:
```suggestion
      if (!href.startsWith('\') && !href.startsWith('/') && postId) {
```

Reply:
I did a benchmark and `String.startsWith()` is 10 times faster. "[url_for()](https://github.com/hexojs/hexo-util/blob/a96335d0f44e0095d6ee036523522e6e880c7b6a/lib/url_for.js#L31)" also can benefit from this.